src/__tests__/selectors.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { UIDisplayData } from '@aics/simularium-viewer';
+import {
+  getPayloadForHideAll,
+} from '../selectors';
+import { ActiveState } from '../constants';
+const mockUIDisplayData: UIDisplayData = [
+  {
+    name: 'agent1',
+    displayStates: [
+      {
+        name: 'state1',
+        id: 'state1',
+        color: '#000000',
+      },
+    ],
+    color: '#000000',
+  },
+  {
+    name: 'agent2',
+    displayStates: [
+      {
+        name: 'state1',
+        id: 'state1',
+        color: '#000000',
+      },
+      {
+        name: 'state2',
+        id: 'state2',
+        color: '#000000',
+      },
+    ],
+    color: '#000000',
+  },
+  {
+    name: 'agent3',
+    displayStates: [],
+    color: '#000000',
+  },
+  {
+    name: 'agent4',
+    displayStates: [
+      {
+        name: 'state1',
+        id: 'state1',
+        color: '#000000',
+      },
+      {
+        name: 'state2',
+        id: 'state2',
+        color: '#000000',
+      },
+      {
+        name: 'state3',
+        id: 'state3',
+        color: '#000000',
+      },
+    ],
+    color: '#000000',
+  },
+  {
+    name: 'agent5',
+    displayStates: [
+      {
+        name: 'state1',
+        id: 'state1',
+        color: '#000000',
+      },
+      {
+        name: 'state2',
+        id: 'state2',
+        color: '#000000',
+      },
+      {
+        name: 'state3',
+        id: 'state3',
+        color: '#000000',
+      },
+    ],
+    color: '#000000',
+  },
+];
+
+const { Active, Inactive, Indeterminate } = ActiveState;
+
+describe('selection composed selectors', () => {
+  describe('getPayloadForHideAll', () => {
+    it('if hiddenState is Inactive, return each agent mapped to an empty array', () => {
+      const payload = getPayloadForHideAll(mockUIDisplayData, Inactive);
+      expect(payload).toEqual({
+        agent1: [],
+        agent2: [],
+        agent3: [],
+        agent4: [],
+        agent5: [],
+      });
+    });
+    it('if hiddenState is Active, return empty object', () => {
+      const payload = getPayloadForHideAll(mockUIDisplayData, Active);
+      expect(payload).toEqual({});
+    });
+    it('if hiddenState is Indeterminate, return empty object', () => {
+      const payload = getPayloadForHideAll(mockUIDisplayData, Indeterminate);
+      expect(payload).toEqual({});
```

Comment:
it's been a while since I worked on this on the front end, but I'm confused why these are the two options. if the object is empty that means it's hiding nothing, ie showing all? but why is an object with agents names mapped to empty strings then the hide all?

Reply:
Yes passing in an empty object means "revealing all" because no agents/displayState will be hidden.

I assume when you say empty string you mean empty array and are referring to this structure in the hide all payloads: ` agent1: [], agent2: [],`  Is that right? 

That is a way to hide everything because if, in selection interface, you pass an empty array for the tags argument in the getIds function, it will will return all the displayStates for that entry.

Reply:
add some of this to the test description